### PR TITLE
Add Mplab XC32 Compiler v2.40

### PR DESCRIPTION
**Pull request by CyberCastle**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

File: Casks/mplab-xc32.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+cask 'mplab-xc32' do
+  version '2.40'
+  sha256 '98eb7107ed13cd08efd5ea1329a1778913ce692225590d7909ec2d62dfccd1be'
+
+  url "https://ww1.microchip.com/downloads/en/DeviceDoc/xc32-v#{version}-full-install-osx-installer.dmg"
+  name 'MPLab XC32 Compiler'
+  homepage 'http://www.microchip.com/pagehandler/en-us/devtools/mplabxc/home.html'
```
Comment (vitorgalvao) [inline Casks/mplab-xc32.rb:7]: 404.

Comment (CyberCastle) [reply Casks/mplab-xc32.rb]: Fixed

**Comment by vitorgalvao** [comment]
Please try to find an [`appcast`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md). There will be something, even if it’s a webpage.

**Comment by CyberCastle** [comment]
This is a development tool that does not have information about future updates, it only has a mere [changelog](http://ww1.microchip.com/downloads/en/DeviceDoc/xc32-v2.40-full-install-release-notes.html). 

According to the instructions in the appcast stanza:

> An appcast can be any URL hosted by the app’s developer that changes every time a new release is out or that contains the version number of the current release (e.g. a download HTML page). Webpages that only change on new version releases are preferred, as are sites that do not contain previous version strings (i.e. avoid changelog pages if the download page contains the current version number but not older ones). (Example: [razorsql.rb](https://github.com/Homebrew/homebrew-cask/blob/645dbb8228ec2f1f217ed1431e188687aac13ca5/Casks/razorsql.rb#L6))

**Comment by vitorgalvao** [comment]
> This is a development tool that does not have information about future updates

Appcasts don’t list future information, they list current information.

> it only has a mere [changelog](http://ww1.microchip.com/downloads/en/DeviceDoc/xc32-v2.40-full-install-release-notes.html).

And a changelog works just fine. There should be a page that doesn’t have a version in the URL but has it in the contents. What’s the download page for this?

**Comment by CyberCastle** [comment]
Is [this](https://www.microchip.com/mplab/compilers), the same as the one indicated in the stanza homepage. So, I add the appcast stanza linked to the described changelog?

**Pull request merged**